inference-engine/src/mkldnn_plugin/nodes/mkldnn_reduce_node.cpp

```diff
@@ -1508,39 +1849,70 @@ void MKLDNNReduceNode::initSupportedPrimitiveDescriptors() {
     }
 }
 
+void MKLDNNReduceNode::prepareParams() {
+    src_dims = getParentEdgesAtPort(REDUCE_DATA)[0]->getMemory().getDesc().getShape().getDims();
+    reduce_axes = raw_axes;
+    if (jit_mode && jit_beyond_5D) {
+        update_src_dims();
+    }
+
+    dst_size = getChildEdgeAt(0)->getMemoryPtr()->GetSize();
+    calc_process_dst_dims();
+    if (jit_mode) {
+        set_reduce_dim_flags();
+    }
+}
+
 void MKLDNNReduceNode::createPrimitive() {
     auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
-    auto &srcDataMemPtr = getParentEdgeAt(REDUCE_DATA)->getMemoryPtr();
-    auto &srcIndexesMemPtr = getParentEdgeAt(REDUCE_INDEXES)->getMemoryPtr();
+    auto &srcMemPtr = getParentEdgeAt(REDUCE_DATA)->getMemoryPtr();
     if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
         IE_THROW() << errorPrefix << " has not allocated destination memory.";
-    if (!srcDataMemPtr || !srcDataMemPtr->GetPrimitivePtr() || !srcIndexesMemPtr || !srcIndexesMemPtr->GetPrimitivePtr())
+    if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
         IE_THROW() << errorPrefix << " has not allocate input memory.";
     if (getSelectedPrimitiveDescriptor() == nullptr)
         IE_THROW() << errorPrefix << " has nullable preferable primitive descriptor";
 
-    auto selectedPD = getSelectedPrimitiveDescriptor();
-    planar_layout = getParentEdgeAt(REDUCE_DATA)->getMemory().getDesc().hasLayoutType(LayoutType::ncsp);
+    if (getParentEdgeAt(REDUCE_DATA)->getMemory().getDesc().hasLayoutType(LayoutType::ncsp)) {
+        layout = ReduceLayoutType::reduce_ncsp;
+    } else if (getParentEdgeAt(REDUCE_DATA)->getMemory().getDesc().hasLayoutType(LayoutType::nspc)) {
+        layout = ReduceLayoutType::reduce_nspc;
+    } else {
+        layout = ReduceLayoutType::reduce_blocked;
+    }
+
+    // hybrid layout: nspc/blocked layout for input and ncsp for output
+    // !keep_dims is needed to avoid hybrid layout for cases eg. (A, B, C, D) reduce to (A, 1, 1, 1)
+    if (!keep_dims && (layout == ReduceLayoutType::reduce_nspc || layout == ReduceLayoutType::reduce_blocked)) {
+        is_hybrid_layout = getChildEdgeAt(REDUCE_DATA)->getMemory().getDesc().hasLayoutType(LayoutType::ncsp);
```

Comment:
Again, is not it better to use `dstMemPtr` here?

Reply:
Fixed. Thanks Maksim!